rpc/trace.go

```diff
@@ -0,0 +1,143 @@
+package rpc
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/NethermindEth/juno/vm"
+)
+
+type TransactionTrace interface {
+	trace()
```

Comment:
We dont need separate types for each transaction.

Let's just use something like

```
type TransactionTrace struct {
	ValidateInvocation    *FunctionInvocation `json:"validate_invocation,omitempty"`
	ExecuteInvocation     *ExecuteInvocation  `json:"execute_invocation,omitempty"`
	FeeTransferInvocation *FunctionInvocation `json:"fee_transfer_invocation,omitempty"`
}
```

and some of the fields can be `nil` according to the transaction type

Reply:
Changed type from interface to struct

Reply:
Can we make them pointers as well? So it is easier to check if they are empty or no

Reply:
Agree, fixed